disperser/dataapi/feed_cache_metrics.go

```diff
@@ -0,0 +1,96 @@
+// This file should have been placed under disperser/dataapi/v2.
+// The reason it's placed here in "dataapi" package is to avoid circular dependency
+// (the "v2" already has dependency on "dataapi").
+// Note the reason there is a "v2" package in the first place is to enable the separation of
+// swagger docs for v1 and v2 APIs.
+
+package dataapi
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+const namespace = "eigenda_dataapi"
+
+type FeedCacheMetrics struct {
+	// Time range metrics
+	cacheTimeRangeSeconds      prometheus.Gauge
+	cacheSegmentStartTimestamp prometheus.Gauge
+	cacheSegmentEndTimestamp   prometheus.Gauge
+
+	// Cache hit metrics
+	cacheHitRatePercent prometheus.Gauge
+}
+
+func NewFeedCacheMetrics(name string, registry *prometheus.Registry) *FeedCacheMetrics {
+	cacheTimeRangeSeconds := promauto.With(registry).NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
```

Comment:
I learned recently that you can use "Subsystem" here. So the namespace is `eigenda` and the subsystem is `dataapi`.

Reply:
Ok, it looks more conventional to use subsystem, and may make it easier to filter metrics on dashboard.